w09h01/test/pgdp/pingutrip/PinguTripTest.java

```diff
@@ -0,0 +1,277 @@
+package pgdp.pingutrip;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class PinguTripTest
+{
+
+	@Nested
+	@DisplayName("readWayPoints(...) tests")
+	class ReadWayPoints
+	{
+		private static final String TEST_FILES_DIRECTORY = "test_txt_files/",
+				NORMAL_TEST_FILE = ReadWayPoints.TEST_FILES_DIRECTORY + "normal.txt",
+				EMPTY = ReadWayPoints.TEST_FILES_DIRECTORY + "empty.txt", OFFICIAL_TEST_FILE = "test_paths/path.txt";
+
+		@BeforeAll
+		public static void setUp()
+		{
+			// create test text files
+			try
+			{
+				new File(ReadWayPoints.TEST_FILES_DIRECTORY).mkdirs();
+				Path normalTest = Path.of(ReadWayPoints.NORMAL_TEST_FILE);
+				if (!Files.exists(normalTest))
+				{
+					normalTest.toFile().createNewFile();
+					Files.writeString(normalTest, """
+							0.0;1.0
+							// comment
+							//7.0;5.0
+							0.0;0.0
+							---
+							5.0;5.0""");
+				}
+				Path empty = Path.of(ReadWayPoints.EMPTY);
+				if (!Files.exists(empty))
+				{
+					empty.toFile().createNewFile();
+					Files.writeString(empty, "");
+				}
+			}
+			catch (Exception e)
+			{
+				System.err.println("error while setting up the ReadWayPoints tests");
+			}
+		}
+
+		@Test
+		@DisplayName(value = "normal test")
+		void testReadWayPointsNormal()
+		{
+			assertIterableEquals(List.of(new WayPoint(0, 1), new WayPoint(0, 0)),
+					PinguTrip.readWayPoints(ReadWayPoints.NORMAL_TEST_FILE).toList(), "normal test");
+		}
+
+		@Test
+		@DisplayName(value = "artemis example")
+		void testReadWayPointsArtemisExample()
+		{
+			assertIterableEquals(List.of(new WayPoint(4.0, 11.5), new WayPoint(19.1, 3.2)),
+					PinguTrip.readWayPoints(ReadWayPoints.OFFICIAL_TEST_FILE).toList(), "example from artemis");
+		}
+
+		@Test
+		@DisplayName(value = "empty test")
+		void testReadWayPointsEmptyStreamOnEmptyStream()
+		{
+			assertEquals(0, PinguTrip.readWayPoints(ReadWayPoints.EMPTY).count(), "empty test");
+		}

```

Comment:
This test is already covered in https://github.com/MaximilianAnzinger/pgdp2223-tests/blob/main/w09h01/test/pgdp/pingutrip/ReadWayPointsTest.java